### DISCUSSION TITLE
skip_api_error in for loop

### DISCUSSION THREAD

**Discussion by ArterialMapping** [mthh/routingpy#61] [Q&A] [answered]
How would I go about using "self.default_skip_api_error:" within a loop? I am unsure as to the correct way to add this into my loop. I am using a CSV file to route 500 different points. 

At a certain coordinate, I get the error "routingpy.exceptions.RouterApiError: 400 ({'error_code': 442, 'error': 'No path could be found for input', 'status_code': 400, 'status': 'Bad Request'})" So I assume this attribute would be the best approach, just not sure the best way to add it in! 

Here is a sample of my loop: 

    for i in range(len(newframe2)):

        X = (newframe2.loc[i, "x"])
        Y = (newframe2.loc[i, "y"])

        coords = [[X, Y], [-77.437391, 37.541556]]

        client = Valhalla(base_url='http://localhost:8002')
        route = client.directions(locations=coords, profile='auto', preference='fastest')

        default_skip_api_error = True

        route_line = gpd.GeoDataFrame(geometry=[LineString(route.geometry)], crs="EPSG:4326").to_crs("EPSG:3857")

        print("Routing ID {}".format(i))
        route_line.to_file(r'{}\Outputs\Route_{}.shp'.format(newpath,i))

**Comment by nilsnolde** [comment, marked as answer]
`default_skip_api_error = True` doesn't do anything. either 
- use valhalla's constructor as show in the docs: https://routingpy.readthedocs.io/en/latest/?badge=latest#valhalla
- use the global `options` object as shown in the readme: https://github.com/gis-ops/routing-py#proxies-rate-limiters-and-api-errors
